wire-ios/Wire-iOS/Sources/UserInterface/Location/AppLocationManager.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreLocation
+
+protocol AppLocationManagerDelegate: AnyObject {
+
+    func didUpdateLocations(_ locations: [CLLocation])
+
+    func didFailWithError(_ error: Error)
+
+    func didChangeAuthorization(status: CLAuthorizationStatus)
+
+}
+
+class AppLocationManager: NSObject {
+
+    // MARK: - Properties
+
+    private let locationManager: CLLocationManager
+    weak var delegate: AppLocationManagerDelegate?
+
+    var authorizationStatus: CLAuthorizationStatus {
+        return locationManager.authorizationStatus
+    }
+
+    // MARK: - Init
+
+    init(locationManager: CLLocationManager) {
```

Comment:
Hm, I don't see the benefit of passing `locationManager` in, as we don't mock it or use different configurations in code.
Can we just instantiate the object internally?

Reply:
Fixed: https://github.com/wireapp/wire-ios/commit/a0f70c0a4d69130334fcb3b07317f00d0bfd66a9